File: escenario1/src/escenario1/PickUpTruck.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package escenario1;

import java.time.LocalDate;

/**
 *
 * @author Dell
 */
public class PickUpTruck extends Vehicle {

    public PickUpTruck(int cylinderCapacity, int modelYear, double valuation) {
        super(cylinderCapacity, modelYear, valuation);
    }
    LocalDate currentDate = LocalDate.now();

    @Override
    double calculateRegistrationCost() {
        double cost = 0;
        int currentYear = currentDate.getYear();
        int vehicleYears = currentYear - modelYear;
        double vehicleCurrentValue = valuation * (1 - (vehicleYears * 0.25));

        cost = vehicleCurrentValue * ((double)cylinderCapacity / 10000);        
        if (cost > 0.4 * valuation) {
            cost = 0.4 * valuation;
        }
        if (cost < 0.1 * valuation) {
            cost = 0.1 * valuation;
        }
        return cost;
    }

}

File: src/Car/PickUpTruck.ts
import { Vehicle } from './Vehicle';

export class PickUpTruck extends Vehicle {
  public constructor(
    public cylinderCapacity: number,
    public modelYear: number,
    public valuation: number,
  ) {
    super(cylinderCapacity, modelYear, valuation);
  }
  private currentDate: Date = new Date();

  calculateRegistrationCost() {
    let cost = 0;
    const currentYear = this.currentDate.getFullYear();
    const vehicleYears = currentYear - this.modelYear;
    const vehicleCurrentValue = this.valuation * (1 - vehicleYears * 0.25);

    cost = vehicleCurrentValue * (this.cylinderCapacity / 10000);
    if (cost > 0.4 * this.valuation) {
      cost = 0.4 * this.valuation;
    }
    if (cost < 0.1 * this.valuation) {
      cost = 0.1 * this.valuation;
    }
    return cost;
  }
}